saltstack/base/salt/dhcp/etc/dhcp/dhclient-enter-hooks

```diff
@@ -69,12 +80,15 @@ make_resolv_conf() {
             for nameserver in ${new_domain_name_servers} ; do
                 REVERSE_DNS=$(dig @${nameserver} -x $IP)
                 log "Reverse DNS by ${nameserver}: ${REVERSE_DNS}"
-                echo "$REVERSE_DNS" | grep -qc "${search}"
+                echo "$REVERSE_DNS" | grep -A 1 "ANSWER SECTION" | grep -qc "${search}"
                 if [[ $? -eq 0 ]]; then
                     set_hostname_from_dns "$REVERSE_DNS" "$IP" "$search"
                     break
                 else
                     log "$nameserver DNS server didn't return valid result for $IP"
+                    # restarting network will force the dhcp client to run again and call make_resolv_conf
+                    call_network_restart
```

Comment:
Can we do a bit differently? I am afraid that it will be stuck in this loop forever if there is some DNS resolution problem, and since the network is continuously restarting, therefore I think ssh-ig into the box is not possible.

Would it be possible to investigate a different approach, a bit less invasive?